### ISSUE TITLE
Documentation tiny-error around async stream example.

### ISSUE THREAD

**Issue by joelapatatechaude** [riemann/riemann#534]
http://riemann.io/howto.html#asynchronous-streams
The core-pool-size is set to 1, but the comment says 4.
In the source code in-line example, it is set to 4.
```clojure
(let [index (index)
      downstream (async-queue!
                    :agg-forwarder        ; A name for the forwarder
                    {:queue-size     1e4  ; 10,000 events max
                     :core-pool-size 1    ; Minimum 4 threads
                     :max-pools-size 100} ; Maxium 100 threads
                    (forward
                      (riemann.client/tcp-client :host "127.0.0.1")))]
````
Additionally, having a core-pool-size set to 1 makes it behave like synchronous in my example (sending messages to slack), even with a max-pool-size set to a large value. It seems like core-pool-size is not a "minimum n threads", but a "maximum number of threads", and it seems like max-pool-size is not relevant. At least that what I see with slack.

Cheers!

**Comment by AeroNotix** [comment]
I followed the example, using :core-pool-size of 1, redeployed and battered riemann, I *seemed* to have synchronous behaviour. @aphyr can you confirm this would be the case?

**Comment by faxm0dem** [comment]
That's **exactly** what I was experiencing the last few weeks (and whining about on IRC)

**Comment by AeroNotix** [comment]
```clojure

(let [host "127.0.0.1"]
  (tcp-server {:host host})
  (udp-server {:host host})
  (ws-server  {:host host})
  (repl-server {:host host}))

(let [index (tap :index (index))
      do-prn (async-queue! :do-prn
                          {:queue-size 1e4
                           :core-pool-size 1
                           :max-pools-size 100}
                          #(do (Thread/sleep 5000)
                               (prn %)))]
  (streams do-prn))
```

But with this example, the client does not block and the events are handled asynchronously.

**Comment by faxm0dem** [comment]
when I debugged this I saw that `riemann executor downstream threads active` was always `1`, even if the `max-pools-size` is `128`

**Comment by joelapatatechaude** [comment]
Ah, I thought it was just a documentation error. Since there are suspicion of bug, I did more test, and I have more to say. First of all, below is essentially a minimal riemann config (which I slightly modify during the test:

```clojure
;; -*- mode: clojure; -*-
;; vim: filetype=clojure
;;@formatter:off
(logging/init)
(instrumentation {:interval 5 :enabled? false})
(require '[clojure.string :refer [join]]
'[riemann.common :refer [time-at]]
'[riemann.slack :refer [slack-escape]]
)
(let [host "0.0.0.0"]
  (tcp-server {:host host})
  (ws-server  {:host host}))

(def credentials {:account "my_account",
                  :token "TOP_SECRET"
                  })

(defn alert-error [e]
  (do (prn e)
      ((slack credentials
              {:username "monitor"
               :channel "#riemann_play"
               }) e)))

(let [ async-alert-error (async-queue!
                          :agg-forwarder
                          {:queue-size 1e4
                           :core-pool-size 1
                           :max-pools-size 100}
                          alert-error)]
  (streams
   (by :host
       (where (service "KABOUM")
                    prn
                    async-alert-error))))
```
And below is a python client:
```python
def send_metric(client, host, service, metric, state):
    print {'host': host, 'service': service, 'metric': metric, 'state': state}
    client.send({'host': host, 'service': service, 'metric': metric, 'state': state})
def send_event():
    client = bernhard.Client()
    send_metric(client, 'host01_play', 'KABOUM', 1234, 'KO')
    send_metric(client, 'host02_play', 'KABOUM', 1234, 'KO')
    send_metric(client, 'host03_play', 'KABOUM', 1234, 'KO')
    send_metric(client, 'host04_play', 'KABOUM', 1234, 'KO')
    send_metric(client, 'host05_play', 'KABOUM', 1234, 'KO')
    send_metric(client, 'host06_play', 'KABOUM', 1234, 'KO')
    send_metric(client, 'host07_play', 'KABOUM', 1234, 'KO')
    send_metric(client, 'host08_play', 'KABOUM', 1234, 'KO')
    send_metric(client, 'host09_play', 'KABOUM', 1234, 'KO')
    send_metric(client, 'host10_play', 'KABOUM', 1234, 'KO')
    send_metric(client, 'host11_play', 'KABOUM', 1234, 'KO')
    send_metric(client, 'host12_play', 'KABOUM', 1234, 'KO')
```

Now I am doing three experiments:
# Experiment 1: all sync.
Instead of calling async-alert-error in the stream processing, I am calling alert-error. This is sync behavior. What I observe is expected:
* the bernhard client talks to riemann in a sync way: each send_metric is blocked until it is treated
* riemann talks to slack (slack is a service similar to IRC) in a sync way because riemann doesn't execute the 'prn' of my async-alert-error one at a time.

# Experiment 2: sync and async
I am now using async-alert-error, with a core-pool-size of 1 and max-pool-size of 100
What I see is that:
* the bernhard client talks to riemann in an async way: all event are send instentaneously, and the python script finishes.
* riemann talks to slack in a sync-alike way, 1 message at a time.

# Experiment 3: full async behavior
I am now using async-alert-error, with a core-pool-size of 20 (and max 100)
* bernhard client returns instantly as expected
* riemann sends all messages to slack instantly, and print them all on the console.

In conclusion, I think things behave as expected, I don't understand what is the purpose of max-pools-size, I tried different values during my experiment, I didn't observe any differences.
I wished the doc was a bit more explicit about max-pool-size, and I recommend to update the example in the howto page: http://riemann.io/howto.html#asynchronous-streams to use a max-pool-size of 4.

Hope it helps!

**Comment by aphyr** [comment]
(What you're seeing is basically really awful threadpool sizing behavior from the jvm cachedthreadpoolexecutor, which only adds threads to the pool when the queue is full.)
